softhddevice.cpp

```diff
@@ -323,95 +292,137 @@ void cSoftHdDevice::HandlePause(void)
  *
  * @param event     The event to process (variant type containing specific event data)
  */
-void cSoftHdDevice::OnEventReceived(const Event& event) {
+void cSoftHdDevice::OnEventReceived(const Event& event)
+{
 	LOGDEBUG("device: received %s", EventToString(event));
 
-	m_pRender->DisplayThreadHalt(); // the display thread needs to be halted first, otherwise a deadlock can occur in WaitForAudioClock()
-	m_pRender->DecodingThreadHalt();
+	if (m_state != DETACHED) {
+		m_pRender->DisplayThreadHalt(); // the display thread needs to be halted first, otherwise a deadlock can occur in WaitForAudioClock()
+		m_pVideoStream->DecodingThreadHalt();
+	}
+
+	bool needsResume = true;
 
 	auto invalid = [this, &event]() {
 		LOGWARNING("device: Invalid event '%s' in state '%s' received", EventToString(event), StateToString(m_state));
 	};
 
 	switch (m_state) {
+		case State::DETACHED:
+			std::visit(overload{
+				[this, &needsResume, &invalid](const PlayEvent&) { invalid(); },
```

Comment:
Only the value used inside the lambda needs to be specified in the captor arguments.
```suggestion
				[&invalid](const PlayEvent&) { invalid(); },
```